app/src/main/java/fr/free/nrw/commons/contributions/ContributionsPresenter.java

```diff
@@ -3,104 +3,156 @@
-                ALL_FIELDS, "", null,
-                ContributionDao.CONTRIBUTION_SORT + "LIMIT "
-                        + (preferredNumberOfUploads>0?preferredNumberOfUploads:100));
+    public void fetchContributions() {
+        Timber.d("fetch  Contributions");
+        LiveData<List<Contribution>> liveDataContributions = repository.fetchContributions();
+        if(null!=lifeCycleOwner) {
+             liveDataContributions.observe(lifeCycleOwner, this::showContributions);
+        }
+
+        if (NetworkUtils.isInternetConnectionEstablished(CommonsApplication.getInstance()) && shouldFetchContributions()) {
+            Timber.d("fetching contributions: ");
+            view.showProgress(true);
+            this.user = sessionManager.getUserName();
+            view.showContributions(Collections.emptyList());
+            compositeDisposable.add(userClient.logEvents(user)
+                    .subscribeOn(ioThreadScheduler)
+                    .observeOn(mainThreadScheduler)
+                    .doOnNext(mwQueryLogEvent -> Timber.d("Received image %s", mwQueryLogEvent.title()))
+                    .filter(mwQueryLogEvent -> !mwQueryLogEvent.isDeleted()).doOnNext(mwQueryLogEvent -> Timber.d("Image %s passed filters", mwQueryLogEvent.title()))
+                    .map(image -> {
+                        Contribution contribution = new Contribution(null, null, image.title(),
+                                "", -1, image.date(), image.date(), user,
+                                "", "", STATE_COMPLETED);
+                        return contribution;
+                    })
+                    .toList()
+                    .subscribe(this::saveContributionsToDB, error -> {
+                        Timber.e("Failed to fetch contributions: %s", error.getMessage());
+                    }));
+        }
     }
 
-    @Override
-    public void onLoadFinished(@NonNull Loader<Cursor> loader, Cursor cursor) {
+    private void showContributions(@NonNull List<Contribution> contributions) {
         view.showProgress(false);
-        if (null != cursor && cursor.getCount() > 0) {
-            view.showWelcomeTip(false);
-            view.showNoContributionsUI(false);
-            view.setUploadCount(cursor.getCount());
-        } else {
+        if (contributions.size() == 0) {
             view.showWelcomeTip(true);
             view.showNoContributionsUI(true);
+        } else {
+            view.showWelcomeTip(false);
+            view.showNoContributionsUI(false);
+            view.setUploadCount(contributions.size());
+            view.showContributions(contributions);
+            this.contributionList.clear();
+            this.contributionList.addAll(contributions);
+        }
+    }
+
+    private void saveContributionsToDB(List<Contribution> contributions) {
+        Timber.e("Fetched: "+contributions.size()+" contributions "+" saving to db");
+        ExecutorUtils.get().submit(() -> repository.save(contributions));
```

Comment:
this could be a Single too, or a Completeable if you don't really care about the return value ehich you don't because it is `void`

Reply:
Yes, I must have missed it somehow, adding it, thanks for pointing it out